module/localisation/OdometryLocalisation/src/OdometryLocalisation.cpp

```diff
@@ -18,35 +16,35 @@ namespace localisation {
     using message::input::Sensors;
     using message::localisation::Field;
     using message::platform::darwin::ButtonLeftDown;
-    using utility::math::matrix::Transform2D;
-    using utility::math::matrix::Transform3D;
     using utility::nusight::graph;
+    using utility::support::Expression;
 
     OdometryLocalisation::OdometryLocalisation(std::unique_ptr<NUClear::Environment> environment)
         : Reactor(std::move(environment)) {
 
         on<Configuration>("OdometryLocalisation.yaml").then([this](const Configuration& config) {
             // Use configuration here from file OdometryLocalisation.yaml
-            localisationOffset = config["localisationOffset"].as<arma::vec>();
+            localisationOffset = config["localisationOffset"].as<Expression>();
         });
 
         on<Trigger<ButtonLeftDown>, Single, With<Sensors>, Sync<OdometryLocalisation>>().then(
             [this](const Sensors& sensors) {
                 NUClear::log("Localisation Orientation reset. This direction is now forward.");
                 emit(std::make_unique<Nod>(true));
-                Transform2D Trw    = Transform3D(convert(sensors.Htw)).projectTo2D();
-                localisationOffset = Trw;
+                Eigen::Affine2d Trw = sensors.Htw.projectTo2D();
+                localisationOffset  = Trw;
             });
 
 
         on<Trigger<Sensors>, Sync<OdometryLocalisation>, Single>().then("Odometry Loc", [this](const Sensors& sensors) {
-            Transform2D Trw = Transform3D(convert(sensors.Htw)).projectTo2D();
-            Transform2D Twr = Trw.i();
+            Eigen::Affine2d Trw = sensors.Htw.projectTo2D();
+            Eigen::Affine2d Twr = Trw.inverse();
 
-            Transform2D state = localisationOffset.localToWorld(Twr);
+            // Local to world transform
+            Eigen::Affine2d state = Twr * localisationOffset;
 
-            auto field        = std::make_unique<Field>();
-            field->position   = Eigen::Vector3d(state.x(), state.y(), state.angle());
+            auto field      = std::make_unique<Field>();
+            field->position = Eigen::Vector3d(state.translation()(0), state.translation()(1), state.rotation().angle());
```

Comment:
Change `Field::position` to a mat33 in `Field.proto` and store state in there directly.